src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -454,6 +454,12 @@ private static IEnumerable<TargetFrameworkInformation> GetTargetFrameworkInforma
             foreach (var item in GetItemByType(items, "TargetFrameworkInformation"))
             {
                 var frameworkString = item.GetProperty("TargetFramework");
+                if (frameworkString.Contains(';'))
+                {
+                    var error = RestoreLogMessage.CreateError(NuGetLogCode.NU1001, string.Format(CultureInfo.CurrentCulture, Strings.Error_InvalidTargetFramework, frameworkString));
```

Comment:
I know this is in draft, and we had chatted about validating the different scenarios. 
If we're going to add something like this, I think static graph might need some changes too.

Reply:
Thanks for letting me know, I'll make sure I update static graph too